File: Medium/reverse_int.py
#max hillyer 6/27/25

class Solution:
    def reverse(self, x: int) -> int:
        thelist = [str(i) for i in str(x)]
        thelist.reverse()
        if '-' in thelist:
            thelist.remove('-')
            print(thelist)
            neg = True
        else:
            neg = False
        final = int("".join(thelist))
        if final > 2**31 - 1 or final < -2**31:
            return 0
        if neg:
            return final - 2*final
        return final

File: Medium/reverse_int.c
//max hillyer 6/28/25

#include <limits.h>
int reverse(int x){
    int new = x;
    long k = 0;
    while (new != 0){
        int n = new%10;
        k *= 10;
        k += n;
        new /= 10;
    }
    if (k > INT_MAX || k < INT_MIN){
        return 0;
    }
    return k;
}

